src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+    ) {
+      return {
+        selectedItems: selectedItems || defaultSelectedItems || [],
+        selectedKeys: getSelectedKeys(selectedItems),
+        disabledKeys: selectedItems
+          ? getDisabledKeys(selectedItems)
+          : getDisabledKeys(propItems),
+        filteredItems: propItems,
+        initialItems: propItems,
+      };
+    }
+    return null;
+  }
+
+  handleItemSelection = (item: T & ComboboxData) => {
+    this.setState(
+      (
+        prevState: ComboboxState<T & ComboboxData>,
+        prevProps: ComboboxProps<T & ComboboxData>,
+      ) => {
+        const {
+          onItemSelectionsChange,
+          onItemSelect,
+          selectedItems: controlledItems,
+        } = prevProps;
+        const { selectedKeys, selectedItems } = prevState;
+        const newSelectedKeys = Object.assign({}, selectedKeys);
+        if (!item.disabled) {
+          if (onItemSelect) {
+            onItemSelect(item.uniqueItemId);
+          }
+          if (!controlledItems) {
+            if (item.uniqueItemId in newSelectedKeys) {
+              delete newSelectedKeys[item.uniqueItemId];
+              const newSelectedItems = selectedItems.filter(
+                (selectedItem) =>
+                  selectedItem.uniqueItemId !== item.uniqueItemId,
+              );
+              if (onItemSelectionsChange) {
+                onItemSelectionsChange(newSelectedItems);
+              }
+              return {
+                selectedKeys: newSelectedKeys,
+                selectedItems: newSelectedItems,
+              };
+            }
+            newSelectedKeys[item.uniqueItemId] = false;
+            const newSelectedItems = selectedItems.concat([item]);
+            if (onItemSelectionsChange) {
+              onItemSelectionsChange(newSelectedItems);
+            }
+            return {
+              selectedKeys: newSelectedKeys,
+              selectedItems: newSelectedItems,
+            };
+          }
+        }
+      },
+    );
+  };
+
+  handleRemoveAllSelections = () => {
+    this.setState(
+      (
+        prevState: ComboboxState<T & ComboboxData>,
+        prevProps: ComboboxProps<T & ComboboxData>,
+      ) => {
+        const { selectedItems } = prevState;
+        const { onItemSelectionsChange, onRemoveAll } = prevProps;
+        const disabledItems = [];
+        const newSelectedKeys: ItemKeys = {};
+        if (onRemoveAll) {
+          onRemoveAll();
+        }
+        // eslint-disable-next-line no-restricted-syntax
+        for (const item of selectedItems) {
+          if (item.disabled) {
+            disabledItems.push(item);
+            newSelectedKeys[item.uniqueItemId] = false;
+          }
+        }
+        if (onItemSelectionsChange) {
+          onItemSelectionsChange(disabledItems);
+        }
+        return { selectedKeys: newSelectedKeys, selectedItems: disabledItems };
+      },
+    );
+  };
+
+  render() {
+    const highlightQuery = (text: string, query: string) => {
```

Comment:
These utils don't have to be inside render unless they access props directly

Reply:
Moved the majority of the utils outside of render where applicable and with slight changes.